### DISCUSSION TITLE
Update value and set `touched` to `false` on a given form field

### DISCUSSION THREAD

**Discussion by alex-vladut** [react-hook-form/react-hook-form#9838] [Q&A] [answered]
I would like to implement a use case where I have 2 input fields: `name` and `code`. As the user enters the `name`, the `code` should be derived from the `name` by only keeping the alphanumeric characters. Once the user touched and changed the `code` themselves, then I don't want to be updated anymore (will keep whatever the user typed). At any time the user could edit the `code` to a different value as well.
The way I would approach this problem looks like this:

```js
  const {
    control,
    setValue,
    getValues,
    formState: { touchedFields },
  } = useFormContext<FormFields>()
  const name = useWatch({
    name: 'name',
    control,
  })

  useEffect(() => {
    if (!name) return

    if (
      !touchedFields.code || // code not yet touched
      !getValues('code') // or it was touched but the field is empty
    ) {
      setValue(
        'code',
      name
        .replace(/ /g, '_') // transform spaces to underscores
        .replace(/[^a-zA-Z0-9\-_]+/g, '') // remove all non-alphanumeric, dashes and underscores
        .toLowerCase(),
        { shouldValidate: true }
      )
      if (touchedFields.code) {
        // resetting touched flag to get the latest updates
        touchedFields.code = false
      }
    }
  }, [name])
```

Did anyone implement something similar? Do you have any better suggestion for how to handle such a use case? Thank you very much 🙏

**Comment by bluebill1049** [comment, marked as answer]
`resetField` api?

**Reply by alex-vladut** [reply]
Thanks, that worked great 👍
